pdata/pmetric/json_test.go

```diff
@@ -164,6 +170,541 @@ func TestMetricsNil(t *testing.T) {
+	datapoint.SetScale(1)
+	datapoint.SetStartTimestamp(internal.NewTimestampFromTime(time.Now()))
+	datapoint.SetFlags(internal.MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NO_RECORDED_VALUE))
+	datapoint.Attributes().UpsertString("string", "value")
+	datapoint.Attributes().UpsertBool("bool", true)
+	datapoint.Attributes().UpsertInt("int", 1)
+	datapoint.Attributes().UpsertDouble("double", 1.1)
+	datapoint.Attributes().UpsertMBytes("bytes", []byte("foo"))
+	datapoint.SetCount(4)
+	datapoint.SetSum(345)
+	datapoint.Positive().SetMBucketCounts([]uint64{1, 1, 2})
+	datapoint.Positive().SetOffset(2)
+	exemplar := datapoint.Exemplars().AppendEmpty()
+	exemplar.SetDoubleVal(99.3)
+	exemplar.SetTimestamp(internal.NewTimestampFromTime(time.Now()))
+	datapoint.SetMin(float64(time.Now().Unix()))
+	traceID := internal.NewTraceID([16]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10})
+	spanID := internal.NewSpanID([8]byte{0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18})
+	exemplar.SetSpanID(spanID)
+	exemplar.SetTraceID(traceID)
+	exemplar.FilteredAttributes().UpsertString("service.name", "testService")
+	datapoint.SetMax(float64(time.Now().Unix()))
+	datapoint.Negative().SetMBucketCounts([]uint64{1, 2, 3})
+	datapoint.Negative().SetOffset(2)
+	datapoint.SetZeroCount(5)
+	datapoint.SetTimestamp(internal.NewTimestampFromTime(time.Now()))
+	return metric
+}
+
+var metricsSummaryOTLPFull = func() Metrics {
+	metric := NewMetrics()
+	rs := metric.ResourceMetrics().AppendEmpty()
+	rs.SetSchemaUrl("schemaURL")
+	// Add resource.
+	rs.Resource().Attributes().UpsertString("host.name", "testHost")
+	rs.Resource().Attributes().UpsertString("service.name", "testService")
+	rs.Resource().SetDroppedAttributesCount(1)
+	// Add InstrumentationLibraryMetrics.
+	m := rs.ScopeMetrics().AppendEmpty()
+	m.Scope().SetName("instrumentation name")
+	m.Scope().SetVersion("instrumentation version")
+	m.SetSchemaUrl("schemaURL")
+	// Add Metric
+	sumData := m.Metrics().AppendEmpty()
+	sumData.SetName("test summary")
+	sumData.SetDescription("test summary")
+	sumData.SetDataType(internal.MetricDataTypeSummary)
+	sumData.SetUnit("unit")
+	datapoint := sumData.Summary().DataPoints().AppendEmpty()
+	datapoint.SetStartTimestamp(internal.NewTimestampFromTime(time.Now()))
+	datapoint.SetFlags(internal.MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NO_RECORDED_VALUE))
+	datapoint.SetCount(100)
+	datapoint.SetSum(100)
+	quantile := datapoint.QuantileValues().AppendEmpty()
+	quantile.SetQuantile(0.5)
+	quantile.SetValue(1.2)
+	datapoint.Attributes().UpsertString("string", "value")
+	datapoint.Attributes().UpsertBool("bool", true)
+	datapoint.Attributes().UpsertInt("int", 1)
+	datapoint.Attributes().UpsertDouble("double", 1.1)
+	datapoint.Attributes().UpsertMBytes("bytes", []byte("foo"))
+	datapoint.SetTimestamp(internal.NewTimestampFromTime(time.Now()))
+	return metric
+}
+
+func TestMetricsSumDataJSONFull(t *testing.T) {
```

Comment:
For all these tests can you also add variation which uses jsonpb marshaler with `OrigName=true` option, `EmitDefaults=true` option and `EnumsAsInts=true` option? The new unmarshaler must work correctly for these cases. I would do a matrix testing for all 8 cases for a combination of these 3 bool options.

Reply:
Thank you for your suggestions,I will do it